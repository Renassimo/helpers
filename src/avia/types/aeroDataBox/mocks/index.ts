export * from './flights';
export * from './airports';
