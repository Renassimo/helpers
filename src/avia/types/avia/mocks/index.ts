export * from './deserializedFlights';
export * from './deserializedAirports';
