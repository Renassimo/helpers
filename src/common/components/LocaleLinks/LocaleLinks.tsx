import Link from 'next/link';

import Box from '@mui/material/Box';
import MuiLink from '@mui/material/Link';

const LocaleLinks = ({
  locale,
  locales,
  href,
}: {
  locale: string;
  locales: string[];
  href: string;
}) => {
  const otherLocales = locales.filter((lcl) => lcl !== locale);

  return (
    <Box display="flex" justifyContent="center" mt={4}>
      {otherLocales.map((lcl) => (
        <Box key={lcl} px={2}>
          <Link href={href} locale={lcl}>
            <MuiLink component="span">{lcl}</MuiLink>
          </Link>
        </Box>
      ))}
    </Box>
  );
};

export default LocaleLinks;
