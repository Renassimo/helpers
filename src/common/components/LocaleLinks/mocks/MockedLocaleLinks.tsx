const MockedLocaleLinks = jest.fn(({ locale, locales, href }) => (
  <div>
    Mocked Locale Links:
    <span>
      {locale} -{' '}
      {locales.map((lcl: string) => (
        <span key={lcl}>{lcl}</span>
      ))}{' '}
      - {href}
    </span>
  </div>
));

export default MockedLocaleLinks;
