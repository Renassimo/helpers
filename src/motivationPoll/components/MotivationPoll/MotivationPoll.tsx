import useMotivationPoll from '@/motivationPoll/hooks/useMotivationPoll';

import PollForm from '@/motivationPoll/components/PollForm';
import Results from '@/motivationPoll/components/Results';

import Box from '@mui/material/Box';

const MotivationPoll = ({ onDownloadPdf }: { onDownloadPdf: () => void }) => {
  const { results } = useMotivationPoll();
  return (
    <Box>
      {results ? <Results onDownloadPdf={onDownloadPdf} /> : <PollForm />}
    </Box>
  );
};

export default MotivationPoll;
