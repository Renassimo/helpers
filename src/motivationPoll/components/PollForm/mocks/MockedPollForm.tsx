const MockedPollForm = jest.fn(() => <div>Mocked Poll Form</div>);

export default MockedPollForm;
