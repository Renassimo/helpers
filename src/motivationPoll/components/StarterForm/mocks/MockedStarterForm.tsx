const MockedStarterForm = jest.fn(() => <div>Mocked Starter Form</div>);

export default MockedStarterForm;
