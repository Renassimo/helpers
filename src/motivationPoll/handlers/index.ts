import getStaticProps from './getStaticProps';

export { getStaticProps };
