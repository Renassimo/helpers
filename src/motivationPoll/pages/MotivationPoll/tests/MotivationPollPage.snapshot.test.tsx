import renderWithTheme from '@/tests/helpers';

import usePdfDownload from '@/common/hooks/usePdfDownload';

import MotivationPoll from '@/motivationPoll/components/MotivationPoll';
import MockedMotivationPoll from '@/motivationPoll/components/MotivationPoll/mocks/MockedMotivationPoll';
import LocaleLinks from '@/common/components/LocaleLinks';
import MockedLocaleLinks from '@/common/components/LocaleLinks/mocks/MockedLocaleLinks';

import { mockedApiData } from '@/types/motivationPoll/mocks';

import MotivationPollPage from '../MotivationPollPage';

jest.mock('@/hooks/common/usePdfDownload');
jest.mock('@/common/components/LocaleLinks');
jest.mock('@/motivationPoll/components/MotivationPoll');

describe('MotivationPollPage snapshot', () => {
  const mockedOnDownloadPdf = jest.fn();
  const mockedPdfRef = null;
  const mockedUsePdfDownload = jest.fn(() => ({
    pdfRef: mockedPdfRef,
    onDownloadPdf: mockedOnDownloadPdf,
  }));
  const mockedLocales = ['ru', 'en'];
  const mockedLocale = 'en';

  beforeEach(() => {
    (usePdfDownload as jest.Mock).mockImplementation(mockedUsePdfDownload);
    (MotivationPoll as jest.Mock).mockImplementation(MockedMotivationPoll);
    (LocaleLinks as jest.Mock).mockImplementation(MockedLocaleLinks);
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { container } = renderWithTheme(
      <MotivationPollPage
        data={mockedApiData}
        locale={mockedLocale}
        locales={mockedLocales}
      />
    );
    // Assert
    expect(mockedUsePdfDownload).toHaveBeenCalledWith(
      'motivation-poll-results'
    );
    expect(container).toMatchSnapshot();
  });
});
