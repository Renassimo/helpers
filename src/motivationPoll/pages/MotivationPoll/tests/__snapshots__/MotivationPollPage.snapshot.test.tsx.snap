// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MotivationPollPage snapshot renders successfully 1`] = `
<div>
  <div>
    <div>
      Mocked Page Template (
      Motivation Test
      ):
      <span>
         - 
      </span>
      <div>
        MockedMotivationPoll
      </div>
      <div>
        Mocked Locale Links:
        <span>
          en
           -
           
          <span>
            ru
          </span>
          <span>
            en
          </span>
           
          - 
          /motivationTest
        </span>
      </div>
    </div>
  </div>
</div>
`;
